debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDefaultValueConverter.java

```diff
@@ -259,6 +260,125 @@ private DateTimeFormatter timestampFormat(int length) {
+        boolean hasDividing = false;
+        for (int i = secondDash + 1; i < s.length(); i++) {
+            if (!Character.isDigit(s.charAt(i))) {
+                dividingStart = i;
+                hasDividing = true;
+                break;
+            }
+        }
+        if (hasDividing) {
+            for (int i = dividingStart; i < s.length(); i++) {
+                if (Character.isDigit(s.charAt(i))) {
+                    break;
+                }
+                dividingEnd = i;
+            }
+        }
+
+        // Parse the time
+        int firstColon = s.indexOf(':', dividingEnd + 1);
+        int secondColon = s.indexOf(':', firstColon + 1);
+        int period = s.indexOf('.', secondColon + 1);
+
+        int year = 0;
+        int month = 0;
+        int day = 0;
+        int hour = 0;
+        int minute = 0;
+        int second = 0;
+
+        // Get the date
+        int len = s.length();
+        boolean parsedDate = false;
+        if (firstDash > 0 && secondDash > firstDash) {
+            year = Integer.parseInt(s.substring(0, firstDash));
+            month = Integer.parseInt(s.substring(firstDash + 1, secondDash));
+            if (hasDividing) {
+                day = Integer.parseInt(s.substring(secondDash + 1, dividingStart));
+            }
+            else {
+                day = Integer.parseInt(s.substring(secondDash + 1, len));
+            }
+
+            if ((month >= 1 && month <= MAX_MONTH) && (day >= 1 && day <= MAX_DAY)) {
+                parsedDate = true;
+            }
+        }
+        if (!parsedDate) {
+            throw new java.lang.IllegalArgumentException("Cannot parse the date from " + s);
+        }
+
+        // Get the time. Hour, minute, second and colons are all optional
+        if (hasDividing && dividingEnd < len - 1) {
+            if (firstColon == -1) {
+                hour = Integer.parseInt(s.substring(dividingEnd + 1, len));
+            }
+            else {
+                hour = Integer.parseInt(s.substring(dividingEnd + 1, firstColon));
+                if (firstColon < len - 1) {
+                    if (secondColon == -1) {
+                        minute = Integer.parseInt(s.substring(firstColon + 1, len));
+                    }
+                    else {
+                        minute = Integer.parseInt(s.substring(firstColon + 1, secondColon));
+                        if (secondColon < len - 1) {
+                            if (period == -1) {
+                                second = Integer.parseInt(s.substring(secondColon + 1, len));
+                            }
+                            else {
+                                second = Integer.parseInt(s.substring(secondColon + 1, period));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        StringBuilder cleanedTimestamp = new StringBuilder();
+        cleanedTimestamp = cleanedTimestamp
```

Comment:
Could this part be consturcted with a single `String.format` for all compoents?

Reply:
Absolutely! Addressed.